Templates/CSharp/Requests/MethodRequestBuilder.cs.tt

```diff
@@ -5,417 +5,119 @@
-private void AddParameterToRequestBuilderConstructorStringBuilder(
-    StringBuilder requestBuilderConstructorParametersStringBuilder,
-    OdcmParameter param,
-    bool optional)
-{
-    var paramTypeString = param.Type.GetTypeString();
-    var paramCheckedCase = param.Name.ToCheckedCase();
-    var paramLowerFirstChar = param.Name.ToLowerFirstChar();
-
-    requestBuilderConstructorParametersStringBuilder.Append(Environment.NewLine);
-    requestBuilderConstructorParametersStringBuilder.Append("            ");
-     
-    if (optional)
-    {
-        if (param.IsCollection)
-        {
-            requestBuilderConstructorParametersStringBuilder.AppendFormat("IEnumerable<{0}> {1} = null,", paramTypeString, paramLowerFirstChar);
-        }
-        else if (param.Type.IsTypeNullable())
-        {
-             requestBuilderConstructorParametersStringBuilder.AppendFormat("{0} {1} = null,", paramTypeString, paramLowerFirstChar);
-        }
-        else
-        {
-             requestBuilderConstructorParametersStringBuilder.AppendFormat("{0}? {1} = null,", paramTypeString, paramLowerFirstChar);
-        }
-    }
-    else
+<#
+    if (isPost)
     {
-        if (param.IsCollection)
+        foreach (var m in methods)
         {
-            paramTypeString = string.Format("IEnumerable<{0}>", paramTypeString);
-        }
-        
-        requestBuilderConstructorParametersStringBuilder.AppendFormat("{0} {1},", paramTypeString, paramLowerFirstChar);
-    }
-}
-
-// Adds the specified parameter to the string builder for the initializations in the request builder constructor body
-private void AddParameterToRequestBuilderInitializerStringBuilder(
-    StringBuilder requestBuilderInitializerStringBuilder,
-    OdcmParameter param)
-{
-    var paramCheckedCase = param.Name.ToCheckedCase();
-    var paramLowerFirstChar = param.Name.ToLowerFirstChar();
-
-    requestBuilderInitializerStringBuilder.Append(Environment.NewLine);
-    requestBuilderInitializerStringBuilder.Append("            ");
-    requestBuilderInitializerStringBuilder.AppendFormat("this.{0} = {1};", paramCheckedCase, paramLowerFirstChar);
-}
-
-// Adds the specified parameter to the string builder for calling the request constructor in the Request() method
-private void AddParameterToRequestConstructorStringBuilder(
-    StringBuilder requestConstructorParametersStringBuilder,
-    OdcmParameter param)
-{
-    var paramCheckedCase = param.Name.ToCheckedCase();
-    
-    requestConstructorParametersStringBuilder.Append(Environment.NewLine);
-    requestConstructorParametersStringBuilder.Append("                ");
-    requestConstructorParametersStringBuilder.AppendFormat("this.{0},", paramCheckedCase);
-}
+            foreach (var p in m.Parameters)
+            {
+#>
+            if (HasParameter("<#=p.Name#>"))
+                request.RequestBody.<#=p.PropertyName#> = GetParameter<<#=p.Type#>>("<#=p.Name#>");
```

Comment:
Also nit: preface functions with this. or ClassName. for static.

---
In reply to: [69045385](https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/pull/24#discussion_r69045385) [](ancestors = 69045385)